packages/governance/src/governor/extensions/governor_settings.cairo

```diff
@@ -0,0 +1,251 @@
+    #[derive(Drop, starknet::Event)]
+    pub struct VotingDelayUpdated {
+        pub old_voting_delay: u64,
+        pub new_voting_delay: u64
+    }
+
+    /// Emitted when `Governor_voting_period` is updated.
+    #[derive(Drop, starknet::Event)]
+    pub struct VotingPeriodUpdated {
+        pub old_voting_period: u64,
+        pub new_voting_period: u64
+    }
+
+    /// Emitted when `Governor_proposal_threshold` is updated.
+    #[derive(Drop, starknet::Event)]
+    pub struct ProposalThresholdUpdated {
+        pub old_proposal_threshold: u256,
+        pub new_proposal_threshold: u256
+    }
+
+    pub mod Errors {
+        pub const INVALID_VOTING_PERIOD: felt252 = 'Invalid voting period';
+    }
+
+    //
+    // Extensions
+    //
+
+    pub impl GovernorSettings<
+        TContractState,
+        +GovernorComponent::HasComponent<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        impl GovernorSettings: HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of GovernorComponent::GovernorSettingsTrait<TContractState> {
+        /// See `GovernorComponent::GovernorSettingsTrait::voting_delay`.
+        fn voting_delay(self: @GovernorComponentState<TContractState>) -> u64 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_voting_delay.read()
+        }
+
+        /// See `GovernorComponent::GovernorSettingsTrait::voting_period`.
+        fn voting_period(self: @GovernorComponentState<TContractState>) -> u64 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_voting_period.read()
+        }
+
+        /// See `GovernorComponent::GovernorSettingsTrait::proposal_threshold`.
+        fn proposal_threshold(self: @GovernorComponentState<TContractState>) -> u256 {
+            let contract = self.get_contract();
+            let this_component = GovernorSettings::get_component(contract);
+
+            this_component.Governor_proposal_threshold.read()
+        }
+    }
+
+    //
+    // External
+    //
+
+    #[embeddable_as(GovernorSettingsAdminImpl)]
+    impl GovernorSettingsAdmin<
+        TContractState,
+        +HasComponent<TContractState>,
+        +GovernorComponent::HasComponent<TContractState>,
+        +GovernorComponent::GovernorCountingTrait<TContractState>,
+        +GovernorComponent::GovernorExecutionTrait<TContractState>,
+        +GovernorComponent::GovernorVotesTrait<TContractState>,
+        +SRC5Component::HasComponent<TContractState>,
+        +Drop<TContractState>
+    > of IGovernorSettingsAdmin<ComponentState<TContractState>> {
+        /// Sets the voting delay.
+        ///
+        /// NOTE: This function does not emit an event if the new voting delay is the same as the
+        /// old one.
```

Comment:
Could be more clear if we move the note here and in the similar functions to the end of the function description